tests/unit_tests/test_all_icons.py

```diff
@@ -0,0 +1,239 @@
+                error_message = f"Color mismatch for icon {icon_name}"
+                acceptable_color_delta = 10
+                if red > 0:
+                    self.assertAlmostEqual(red,
+                                           average_color[0],
+                                           delta=acceptable_color_delta,
+                                           msg=error_message)
+                if green > 0:
+                    self.assertAlmostEqual(green,
+                                           average_color[1],
+                                           delta=acceptable_color_delta,
+                                           msg=error_message)
+                if blue > 0:
+                    self.assertAlmostEqual(blue,
+                                           average_color[2],
+                                           delta=acceptable_color_delta,
+                                           msg=error_message)
+
+    def test_all_icon_sets_endpoint(self):
+        """
+        Checking that the endpoint /iconsets returns all available icon sets
+        """
+        response = self.app.get(f"{ROUTE_PREFIX}/", headers={"Origin": "map.geo.admin.ch"})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content_type, "application/json")
+        icon_sets_from_endpoint = response.json
+        self.assertEqual(len(icon_sets_from_endpoint), len(self.all_icon_sets))
+        for icon_set in icon_sets_from_endpoint:
+            self.assertIn('name', icon_set)
+            self.assertTrue(icon_set['name'] in self.all_icon_sets)
+            self.assertIn('colorable', icon_set)
+            self.assertIn('icons_url', icon_set)
+
+    def test_all_icon_sets_metadata_endpoint(self):
+        """
+        Checking that the endpoint /iconsets/{icon_set_name} returns all relevant information
+        about an icon set, and that the icon URL given provides all available icons
+        """
+        for icon_set_name in self.all_icon_sets:
+            response = self.app.get(f"{ROUTE_PREFIX}/{icon_set_name}",
+                                    headers={"Origin": "map.geo.admin.ch"})
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(response.content_type, "application/json")
+            icon_set_metadata = response.json
+            self.assertIn('name', icon_set_metadata)
+            self.assertEqual(icon_set_name, icon_set_metadata['name'])
+            self.assertIn('colorable', icon_set_metadata)
+            self.assertIn('icons_url', icon_set_metadata)
+            icons_response = self.app.get(icon_set_metadata['icons_url'],
+                                          headers={"Origin": "map.geo.admin.ch"})
+            self.assertEqual(icons_response.status_code, 200)
+            self.assertEqual(icons_response.content_type, "application/json")
+            self.assertEqual(len(icons_response.json), len(self.all_icon_sets[icon_set_name]))
```

Comment:
This could be improved by using subtest which gives then a better error message including which icon set name failed:

```python
for icon_set_name in self.all_icon_sets:
  with self.subTest(icon_set_name=icon_set_name):
     ...
```
https://docs.python.org/3.7/library/unittest.html#distinguishing-test-iterations-using-subtests